runtime/complex.go

```diff
@@ -68,5 +69,80 @@ func complexRepr(f *Frame, o *Object) (*Object, *BaseException) {
 }
 
 func initComplexType(dict map[string]*Object) {
+	ComplexType.slots.Eq = &binaryOpSlot{complexEq}
+	ComplexType.slots.GE = &binaryOpSlot{complexGE}
+	ComplexType.slots.GT = &binaryOpSlot{complexGT}
+	ComplexType.slots.LE = &binaryOpSlot{complexLE}
+	ComplexType.slots.LT = &binaryOpSlot{complexLT}
+	ComplexType.slots.NE = &binaryOpSlot{complexNE}
 	ComplexType.slots.Repr = &unaryOpSlot{complexRepr}
 }
+
+func complexEq(f *Frame, v, w *Object) (*Object, *BaseException) {
+	e, ok := complexCompare(toComplexUnsafe(v), w)
+	if !ok {
+		return NotImplemented, nil
+	}
+	return GetBool(e).ToObject(), nil
+}
+
+func complexGE(f *Frame, v, w *Object) (*Object, *BaseException) {
+	return comparisonNotSuported(f, w)
+}
+
+func complexGT(f *Frame, v, w *Object) (*Object, *BaseException) {
+	return comparisonNotSuported(f, w)
+}
+
+func complexLE(f *Frame, v, w *Object) (*Object, *BaseException) {
+	return comparisonNotSuported(f, w)
+}
+
+func complexLT(f *Frame, v, w *Object) (*Object, *BaseException) {
+	return comparisonNotSuported(f, w)
+}
+
+func complexNE(f *Frame, v, w *Object) (*Object, *BaseException) {
+	e, ok := complexCompare(toComplexUnsafe(v), w)
+	if !ok {
+		return NotImplemented, nil
+	}
+	return GetBool(!e).ToObject(), nil
+}
+
+func complexCompare(v *Complex, w *Object) (bool, bool) {
+	lhsr := real(v.Value())
+	rhs, ok := complexCoerce(w)
+	if !ok {
+		return false, false
+	}
+	return lhsr == real(rhs) && imag(v.Value()) == imag(rhs), true
+}
+
+func comparisonNotSuported(f *Frame, w *Object) (*Object, *BaseException) {
```

Comment:
Maybe call this complexCompareNotSupported or something, since comparisonNotSupported is pretty generic.